### PR TITLE
[TextFields] Capture snapshots outside of view

### PR DESCRIPTION
**Pull request by andrewoverton**
This PR makes it so that textcontrol snapshot test cases validate UIView snapshots of text controls inside container views as opposed to validating just UIView snapshots of text controls themselves. This is so that when floating labels go outside the bounds, as they partially do with the outline style, that can be captured in the snapshot.

It also (I think) gets rid of some flakiness in the MDCOutlinedTextFieldSnapshotTests with the changes to MDCTextControlStyleOutlined.

Here's an example snapshot diff:

Screen Shot 2019-11-04 at 4 48 31 PM

### CODE REVIEW

File: components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m [hunk 1]
```diff
@@ -137,8 +137,13 @@ - (void)applyStyleTo:(UIView *)view
                                                               containerHeight:containerHeight
                                                                     lineWidth:outlineLineWidth
                                                               isLabelFloating:isLabelFloating];
+
+  [CATransaction begin];
+  [CATransaction setValue:(id)kCFBooleanTrue forKey:kCATransactionDisableActions];
```
Comment (yarneo) [inline components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m]: can't you use setDisableActions API?

File: components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m [hunk 2]
```diff
@@ -137,8 +137,13 @@ - (void)applyStyleTo:(UIView *)view
                                                               containerHeight:containerHeight
                                                                     lineWidth:outlineLineWidth
                                                               isLabelFloating:isLabelFloating];
+
+  [CATransaction begin];
```
Comment (yarneo) [inline components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m:141]: how come we didnt need this before and now we do?

File: components/TextFields/tests/snapshot/MDCBaseTextFieldTestsSnapshotTestHelpers.m [hunk 3]
```diff
@@ -27,6 +32,43 @@
...
+
++ (MDCOutlinedTextField *)createOutlinedTextField {
+  CGRect frame = [MDCBaseTextFieldTestsSnapshotTestHelpers defaultTextFieldFrame];
+  MDCOutlinedTextField *textField = [[MDCOutlinedTextField alloc] initWithFrame:frame];
+  [MDCBaseTextFieldTestsSnapshotTestHelpers prepareTextFieldForSnapshotTesting:textField];
+  return textField;
+}
+
++ (CGRect)defaultTextFieldFrame {
+  return CGRectMake(0, 0, 200, 60);
```
Comment (yarneo) [inline components/TextFields/tests/snapshot/MDCBaseTextFieldTestsSnapshotTestHelpers.m:58]: would this be able to capture multi-lines? should this use sizeToFit?

File: components/TextFields/tests/snapshot/MDCTextControlSnapshotTestHelpers.m [hunk 4]
```diff
@@ -26,6 +30,27 @@ @implementation MDCTextControlSnapshotTestHelpers
 
 #pragma mark Validation
 
++ (void)addTextControlToKeyWindow:(UIView<MDCTextControl> *)textControl {
+  // Add the text control to a container view. This is the view we snapshot. Snapshotting this view
+  // instead of the text control allows us to see the floating label go outside the bounds for the
+  // outlined style.
+  UIView *container = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 1, 1)];
```
Comment (yarneo) [inline components/TextFields/tests/snapshot/MDCTextControlSnapshotTestHelpers.m]: why give it a frame so small?

File: components/TextFields/tests/snapshot/MDCTextControlSnapshotTestHelpers.m [hunk 5]
```diff
@@ -48,6 +73,15 @@ + (void)forceLayoutOfTextControl:(UIView<MDCTextControl> *)textControl {
   [textControl sizeToFit];
   [textControl setNeedsLayout];
   [textControl layoutIfNeeded];
+  UIView *container = [textControl superview];
+  CGRect textControlFrame = textControl.frame;
+  CGFloat textControlWidth = CGRectGetWidth(textControlFrame);
+  CGFloat textControlHeight = CGRectGetHeight(textControlFrame);
+  CGRect newContainerFrame = CGRectMake(0, 0, textControlWidth + 30, textControlHeight + 30);
```
Comment (yarneo) [inline components/TextFields/tests/snapshot/MDCTextControlSnapshotTestHelpers.m]: what is the 30 and 15 paddings used for?

File: components/TextFields/tests/snapshot/MDCBaseTextFieldTestsSnapshotTestHelpers.m (hunk 3 above)
Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCBaseTextFieldTestsSnapshotTestHelpers.m:58]: `-sizeToFit` is called on the textfield before validation. This is just an initial frame.

File: components/TextFields/tests/snapshot/MDCTextControlSnapshotTestHelpers.m (hunk 4 above)
Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCTextControlSnapshotTestHelpers.m]: For some reason while working on it I was getting some error about zero width layers, so I made it non zero. But I just looked again, and apparently something has changed, and it's no longer an issue. So now I just initialize the view with CGRectZero.

File: components/TextFields/tests/snapshot/MDCTextControlSnapshotTestHelpers.m (hunk 5 above)
Comment (andrewoverton) [reply components/TextFields/tests/snapshot/MDCTextControlSnapshotTestHelpers.m]: 15 is just an arbitrary padding. I am making the code more clear.

File: components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m (hunk 2 above)
Comment (andrewoverton) [reply components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m:141]: It was animating without me telling it to. It was getting "caught in an animation," which was leading to flakiness in the snapshot tests. For now I don't want the outlined style to animate at all.

File: components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m (hunk 1 above)
Comment (andrewoverton) [reply components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m]: Will do!

Comment (andrewoverton) [reply components/TextFields/src/ContainedInputView/private/MDCTextControlStyleOutlined.m]: Done.

**Review by yarneo — APPROVED** [review]

**Pull request merged**